pkg/lifecycle/render/helm/dependency.go

```diff
@@ -21,38 +23,41 @@ func (f *LocalTemplater) addDependencies(
 	helmHome string,
 	chartRoot string,
 	asset api.HelmAsset,
-) error {
+) (depPaths []string, err error) {
 	for _, dependency := range dependencies {
 		if dependency.Repository != "" {
 			repoURL, err := url.Parse(dependency.Repository)
 			if err != nil {
-				return errors.Wrapf(err, "parse dependency repo %s", dependency.Repository)
+				return depPaths, errors.Wrapf(err, "parse dependency repo %s", dependency.Repository)
 			}
 			if repoURL.Scheme == "file" {
-				if err := f.getLocalDependency(dependency.Repository, chartRoot, asset, helmHome); err != nil {
-					return errors.Wrapf(err, "get local dep %s", dependency.Repository)
+				depPath, err := f.getLocalDependency(dependency.Repository, chartRoot, asset, helmHome)
+				if err != nil {
+					return depPaths, errors.Wrapf(err, "get local dep %s", dependency.Repository)
 				}
+				depPaths = append(depPaths, depPath)
 			} else {
 				repoName := strings.Split(repoURL.Hostname(), ".")[0]
 				if err := f.Commands.RepoAdd(repoName, dependency.Repository, helmHome); err != nil {
-					return errors.Wrapf(err, "add helm repo %s", dependency.Repository)
+					return depPaths, errors.Wrapf(err, "add helm repo %s", dependency.Repository)
 				}
 			}
 		}
 	}
 
-	return nil
+	return depPaths, nil
```

Comment:
returning the paths of the local dependencies fetched here so that they can be cleaned up later during `cleanUpAndOutputRenderedFiles`